core/src/main/java/com/linecorp/armeria/server/logging/structured/ApacheThriftRpcStructuredLog.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    /**
+     * Constructs {@link ApacheThriftRpcStructuredLog} from {@link RequestContext} and {@link ResponseLog}.
+     * Can be used as {@link StructuredLogBuilder}.
+     */
+    public ApacheThriftRpcStructuredLog(RequestContext reqCtx, ResponseLog resLog) {
+        super(resLog);
+
+        RequestLog req = resLog.request();
+
+        RpcRequest rpcRequest = req.attr(RequestLog.RPC_REQUEST).get();
+        if (rpcRequest == null) {
+            // Request might be responded as error before reading arguments.
+            thriftServiceName = null;
+            thriftMethodName = null;
+            thriftCall = null;
+            thriftReply = null;
+            return;
+        }
+
+        if (!(rpcRequest instanceof ThriftCall)) {
+            throw new IllegalArgumentException(
+                    "expected ThriftCall instance for " + RequestLog.RPC_REQUEST + " but was " + rpcRequest);
+        }
+        ThriftCall thriftCall = (ThriftCall) rpcRequest;
+
+        thriftServiceName = thriftIfaceClassSuffixRegexp.matcher(thriftCall.serviceType().getCanonicalName())
+                                                        .replaceAll("");
+
+        thriftMethodName = thriftCall.method();
+        this.thriftCall = (ApacheThriftCall) req.attr(RequestLog.RAW_RPC_REQUEST).get();
+        thriftReply = (ApacheThriftReply) resLog.attr(ResponseLog.RAW_RPC_RESPONSE).get();
+    }
+
+    /**
+     * Returns the fully qualified Thrift service name which is associated to the log.
+     * @return fully qualified Thrift service name
+     */
+    @JsonProperty
+    public String thriftServiceName() {
+        return thriftServiceName;
+    }
+
+    /**
+     * Returns the Thrift method name which was called in the context of the log.
+     * @return Thrift method name
+     */
+    @JsonProperty
+    public String thriftMethodName() {
+        return thriftMethodName;
+    }
+
+    /**
+     * Returns the {@link ApacheThriftCall} object which includes Thrift call information of the log.
+     * @return an instance of {@link ApacheThriftCall} which is associated to the log
+     */
+    @JsonProperty
+    public ApacheThriftCall thriftCall() {
+        return thriftCall;
+    }
+
+    /**
+     * Returns the {@link ApacheThriftReply} object which includes Thrift reply information of the log.
+     * @return an instance of {@link ApacheThriftReply} which is associated to the log
+     */
+    @JsonProperty
+    public ApacheThriftReply thriftReply() {
+        return thriftReply;
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+                          .add("timestampMillis", String.format(
+                                  "%d(%s)", timestampMillis(), DateTimeFormat.formatEpoch(timestampMillis())))
+                          .add("responseTimeNanos", String.format(
+                                  "%d(%s)", responseTimeNanos(), UnitFormatter.elapsed(responseTimeNanos())))
```

Comment:
I'm not sure if it's a good idea to use `String.format()` for such a trivial case. How about just using the plus operators?